src/main/java/org/infinispan/tutorial/embedded/OpenWeatherMapService.java

```diff
@@ -50,9 +50,12 @@ private Document fetchData(String location) {
    }
 
    @Override
-   protected LocationWeather fetchWeather(String location) {
+   protected LocationWeather fetchWeather(String location) throws Exception {
 
       Document dom = fetchData(location);
+      if (dom == null) {
+         throw new Exception("Unable to reach or get response from open weather service with given OWMAPIKEY : please try with valid OWMAPIKEY or use Random Weather Service");
```

Comment:
I'd maybe change this Exception to `RuntimeException` so that you don't have to propagate it via the API generically.

Reply:
Thanks for the review. I'll update this week-end.

Reply:
Agreed, this way you don't have to declare throws Exception everywhere.